src/v/utils/retry.h

```diff
@@ -58,7 +62,13 @@ ss::futurize_t<std::invoke_result_t<Func>> retry_with_backoff(
                              &backoff_policy,
                              &promise,
                              max_retries,
+                             as,
                              base_backoff]() mutable {
+                     if (as.has_value() && as->get().abort_requested()) {
+                         promise.set_exception(ss::abort_requested_exception());
```

Comment:
`ss::abort_requested_exception` isn't the only type of exception that an `abort_source` can be aborted with.

I can't think of a cleaner way to extract the exception:
```suggestion
                         try {
                             as->get().check();
                         } catch (std::exception const& e) {
                             promise.set_exception(e);
                         }
```